web/modules/custom/weather_block/weather_block.install

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * @file
+ * Install, update, and uninstall functions for the weather_block module.
+ */
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Database\Schema;
+
+/**
+ * Implements hook_schema().
+ */
+function weather_block_schema() {
+  $schema['weather_block_cities'] = [
+    'description' => 'Stores data for each city.',
+    'fields' => [
+      'id' => [
+        'type' => 'serial',
+        'not null' => TRUE,
+        'description' => 'Primary Key: Unique ID for the city.',
+      ],
+      'name' => [
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => TRUE,
+        'description' => 'Name of the city.',
+      ],
+      'data' => [
+        'type' => 'text',
+        'not null' => TRUE,
+        'description' => 'City data.',
+      ],
+    ],
+    'primary key' => ['id'],
+  ];
+  $schema['weather_block_city_users'] = [
+    'description' => 'Stores relationships between users and cities.',
+    'fields' => [
+      'id' => [
+        'type' => 'serial',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'description' => 'Primary Key: Unique ID for the relationship.',
+      ],
+      'user_id' => [
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+        'description' => 'ID of the user.',
+      ],
+      'city_name' => [
+        'type' => 'varchar',
+        'length' => 255,
+        'not null' => TRUE,
+        'description' => 'Name of the city.',
+      ],
+    ],
+    'indexes' => [
+      'user_id' => ['user_id'],
+      'city_name' => ['city_name'],
+    ],
+    'foreign keys' => [
+      'user_id' => [
+        'table' => 'users',
+        'columns' => ['user_id' => 'id'],
+      ],
+    ],
+    'primary key' => ['id'],
+  ];
+
+  return $schema;
+
+}
+
+/**
+ * Implements hook_update_N().
+ *
+ * Create the weather_block_cities, weather_block_city_users table.
+ */
+function weather_block_update_91001(): void {
```

Comment:
Not sure this update Number is following standards. But whatever, just FYI no need to change

Reply:
You added a table after all changes, and haven't added new update_hook for that new table.